---
layout: paper
title: A comparative study of topology-based pathway enrichment analysis methods
image: /assets/images/papers/reviewGSEA.png
authors: Jing Ma, Ali Shojaie,  George Michailidis 
year: 2019
shortref: Ma et al. (2019). BMC Bioinformatics.
journal: "BMC Bioinformatics"
pdf: /assets/pdfs/papers/2019-Ma-NetGSA.pdf 
doi: 10.1186/s12859-019-3146-1
github: https://github.com/drjingma/NetGSAreview
type:
---

# Abstract


#### Background

Pathway enrichment analysis is extensively used in the analysis of Omics data for gaining biological insights into the functional roles of pre-defined subsets of genes, proteins and metabolites. A large number of methods have been proposed in the literature for this task. The vast majority of these methods use as input expression levels of the biomolecules under study together with their membership in pathways of interest. The latest generation of pathway enrichment methods also leverages information on the topology of the underlying pathways, which as evidence from their evaluation reveals, lead to improved sensitivity and specificity. Nevertheless, a systematic empirical comparison of such methods is still lacking, making selection of the most suitable method for a specific experimental setting challenging. This comparative study of nine network-based methods for pathway enrichment analysis aims to provide a systematic evaluation of their performance based on three real data sets with different number of features (genes/metabolites) and number of samples. 

#### Results

The findings highlight both methodological and empirical differences across the nine methods. In particular, certain methods assess pathway enrichment due to differences both across expression levels and in the strength of the interconnectedness of the members of the pathway, while others only leverage differential expression levels. In the more challenging setting involving a metabolomics data set, the results show that methods that utilize both pieces of information (with NetGSA being a prototypical one) exhibit superior statistical power in detecting pathway enrichment. 

#### Conclusion

The analysis reveals that a number of methods perform equally well when testing large size pathways, which is the case with genomic data. On the other hand, NetGSA that takes into consideration both differential expression of the biomolecules in the pathway, as well as changes in the topology exhibits a superior performance when testing small size pathways, which is usually the case for metabolomics data.

**Keywords**: Pathway enrichment analysis, Pathway topology, Type I error, Power, Differential network biology

